src/common/components/select-repository-list/index.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component, PropTypes } from 'react';
```

Comment:
Most of this component has been copied from `src/common/components/repository-list.js`. I've separated it out into its own page and its own route to make merging with work @bartaz is doing easier.